tests/explicit-nulls/neg/i21629/J.java

```diff
@@ -0,0 +1,50 @@
+package javax.annotation;
+import java.util.*;
+
+public class J {
+
+    private static String getK() {
+        return "k";
+    }
+
+    @Nullable
+    public static final String k = getK();
+
+    @Nullable
+    public static String l = "l";
+
+    @Nullable
+    public final String m = null;
+
+    @Nullable
+    public String n = "n";
+
+    @Nullable
+    public static final String f(int i) {
+        return "f: " + i;
+    }
+
+    @Nullable
+    public static String g(int i) {
+        return "g: " + i;
+    }
+
+    @Nullable
+    public String h(int i) {
+        return "h: " + i;
+    }
+
+    @Nullable
+    public <T> String[] genericf(T a) {
+        String[] as = new String[1];
+        as[0] = "" + a;
+        return as;
+    }
+
+    @Nullable
```

Comment:
We should test annotations at parameter positions as well: `f(@NotNull String s)`, `g(@Nullable String s)`

Reply:
It seems like `@NotNull` is currently not working in parameter position.

Also, I am not sure about how to test `@Nullable` in parameter position, since we should be testing that the parameter is a `(String | Null)` instead of a `String?`.

I discussed with @olhotak, and one possible way is through overriding, where the following code would fail
```
public class JavaClass {
    public int f(@Nullable String s)
}
class ScalaClass extends JavaClass {
    override def f(s: String): Int
}
```
However we handle overriding logic through a special case for explicit nulls, so this currently doesn't fail. We might have to change the behavior in some other files.

Reply:
I discussed offline with @noti0na1 . We agreed that we should try to remove the special cases in the overriding logic for explicit nulls. Flexible types would make them unnecessary. I've create a new issue @22071 about this.

After that's done, it should be possible to create such tests here using overriding.